lib/sprockets/uglifier_compressor.rb

```diff
@@ -47,14 +47,24 @@ def call(input)
       if Autoload::Uglifier::VERSION.to_i < 2
         raise "uglifier 1.x is no longer supported, please upgrade to 2.x"
       end
-      @uglifier ||= Autoload::Uglifier.new(@options.merge({ source_filename: input[:filename] }))
+      uglifier = Autoload::Uglifier.new(@options.merge(input_options(input)))
```

Comment:
You didn't mention why you're removing the ivar memoization... I guess it was invalidated when we started passing the filename in? Is it slow enough that this should concern us, or was it previously a premature optimization?